src/protocol/message_layer/StreamMessage.js

```diff
@@ -89,19 +85,63 @@ export default class StreamMessage {
         return !!this.getParsedContent()[BYE_KEY]
     }
 
-    static create(messageIdArgsArray, prevMessageRefArgsArray, contentType, content, signatureType, signature) {
+    static create(messageIdArgsArray, prevMessageRefArgsArray, contentType, encryptionType, content, signatureType, signature) {
         const C = StreamMessage.latestClass
-        return new C(messageIdArgsArray, prevMessageRefArgsArray, contentType, content, signatureType, signature)
+        return new C(messageIdArgsArray, prevMessageRefArgsArray, contentType, encryptionType, content, signatureType, signature)
+    }
+
+    static validateContentType(contentType) {
+        if (!Object.values(StreamMessage.CONTENT_TYPES).includes(contentType)) {
```

Comment:
simpler and faster

```suggestion
        if (!StreamMessage.CONTENT_TYPES[contentType]) {
```

Reply:
`contentType` is the value here (27, 28, 29 or 30) and not the key (MESSAGE, GROUP_KEY_REQUEST, etc). So `StreamMessage.CONTENT_TYPES[contentType]` is always false.

Reply:
Whoops, of course, my bad! In this case the comment should be:

Would be nice to do a fast key-based lookup here instead of extracting values to an array and searching it for an item - on every message.

Since the set of valid values is static, you could generate (once) an object with (27, 28, 29, 30) as keys, or use the Set object, then use that for fast validation of the value.